metabase/metabase
# FE - Allow to use in dashboard filters both pre- and post- last aggregation columns

Closes #46522
Closes #46915

### Description

This PR makes it possible to choose pre- and post- last aggregation columns in the UI (in both dashcard filter and click behavior mappings):
- This PR is targeted against an integration branch
- More tests will follow




### How to verify

To check integration with BE changes: `git merge origin/mblib-dashboard-filters-multiple-stage`

---

For any question (or model or metric), it should be possible to map dashboard filters (and click behavior when target destination is a saved question) to the same set of columns as the filter modal in chill mode shows for a given question (depending on dashboard filter type though). Drilling through a question with filter applied should work

Example:

1. New > Question > Custom column > Aggregate > Breakout > Custom column
2. Save and add to dashboard
3. Edit dashboard
4. It should be possible to map columns created by aggregations, breakouts, and custom column to dashboard filters (but column type and filter type need to match)
5. Create a parameter and mapping, save dashboard
6. Set parameter value
7. Click question title to drill through to it. Verify that filter is applied.

### Demo

Questions/Models used in the demo are the same as in unit tests (`createComplex1StageQuery` and `createComplex2StageQuery`).

https://github.com/user-attachments/assets/699bfb2d-695d-405d-ab33-129c3c0ba1e2
